lib/AppInfo/SearchProvider.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Notes\AppInfo;
+
+use OCA\Notes\Service\Note;
+use OCA\Notes\Service\NotesService;
+use OCA\Notes\Service\Util;
+
+use OCP\IUser;
+use OCP\IURLGenerator;
+use OCP\Search\IProvider;
+use OCP\Search\ISearchQuery;
+use OCP\Search\SearchResult;
+use OCP\Search\SearchResultEntry;
+
+class SearchProvider implements IProvider {
+
+	/** @var Util */
+	private $util;
+	/** @var NotesService */
+	private $notesService;
+	/** @var IURLGenerator */
+	private $url;
+
+	public function __construct(
+		Util $util,
+		NotesService $notesService,
+		IURLGenerator $url
+	) {
+		$this->util = $util;
+		$this->notesService = $notesService;
+		$this->url = $url;
+	}
+
+
+	public function getId(): string {
+		return Application::APP_ID;
+	}
+
+	public function getName(): string {
+		return $this->util->l10n->t('Notes');
+	}
+
+	public function getOrder(string $route, array $routeParameters): int {
+		if (strpos($route, 'files' . '.') === 0) {
+			return 25;
+		} elseif (strpos($route, Application::APP_ID . '.') === 0) {
+			return -1;
+		}
+		return 4;
```

Comment:
```suggestion
		if (strpos($route, Application::APP_ID . '.') === 0) {
			return -1;
		}
		return 40;
```

Reply:
So yeah, as korel already mentioned. Notes are files, but the notes app can present them better. So it should be the prefered app when you search anywhere over files, hence the `4` in general, because it should be before files which has 5. Then again on files you might prefer the files search (or not) which is why 25 was picked there, and -1 when in notes.